### PR TITLE
🔎 SEO - Fix SEO on Homepage + Events 

### LINKED ISSUE #2248: 🔎 SEO - Make SEO Fixes
cc @jeoffreyfischer @wicksipedia @camillars 

As per Jeoffrey's conversation with Matt W, we have had a downturn in leads coming from Google. 

MicrosoftTeams-image (8)
**Figure: Low search ranking on Google (14th on the list)**

```[tasklist]
### Tasks
- [ ] Run Semrush on the site, fix SEO issues
- [ ] Check all tags are performing
- [x] Fix Canonical URLs
- [x] Add codeauditor + redirects to robots
- [ ] Fix images on /events on SharePoint
```

### PR DESCRIPTION
**Pull request by Harry-Ross**
* Converted imported components in `blocks-renderer.tsx` and `mdxComponentRenderer.tsx` to use `next/dynamic` as per https://www.ssw.com.au/rules/dynamically-load-components/
* 🎉 Added React Query to `/events` to allow loading more events if required, reduces the amount of data shipped to the page, better caching functionality, now has infinite scroll
  * Added pagination to events API routes 
* Added `useReportWebVitals` to the layout to report web vitals (e.g. FCP, CLS) to App Insights
* Added `memo` to component that was unecessarily rerendering on homepage (flickering)
* Restructured carousel components to dynamic import `react-responsive-carousel` and made base size of carousel div to avoid layout shifts when images load in
* Used image "sizes" (https://nextjs.org/docs/pages/api-reference/components/image#sizes) to reduce the file size of images, added to some MDX components for better fine grained control
* Changed service cards to not use `rich-text`, as its just a string
* Fixed references to events page with old query string (/events?tech=all&type=all)
* General cleanup i.e. adding props to elements, changing from function model to React component model
* Changed CSS background images on pages to use `getImageProps`, which means they can use the features of `next/image`
* Added `aria-label` to social icons
* Fixed sitemap to not reference the v1 site, got useful non-v3 URLs from sitemap and made an additional site
* Fixed sitemap to use real index, not `/home`
* Upgraded `ssw.megamenu` package from 4.2.2 to 4.5.3
* Added caching for API route `/api/get-megamenu`
* Changed from query string manual schema validation to using `yup`
* Fixed canonical URL for `/clients` pages
* Added priority loading for LCP elements (i.e. carousel)
* Fixed hydration issue on `/offices` pages with testimonials 

- Affected routes: `/`, `/events`

- Fixed #2248

- [ ] If adding a new page, I have followed the [📃 New Webpage](https://github.com/SSWConsulting/SSW.Website/issues/new?assignees=&labels=&projects=&template=new_webpage.yml&title=%F0%9F%93%84+%7B%7B+TITLE+%7D%7D+) issue template

- [x] Include done video or screenshots

Before (mobile):
image

After (mobile):
image

Before (desktop):
image

After (desktop):
image

**Load more on `/events`**
image

### CODE REVIEW

File: components/blocks/carousel.tsx [hunk 1]
```diff
@@ -55,20 +61,35 @@ export const Carousel = ({ data }) => {
...
 
-const createCarouselItemImage = (props, index: React.Key, carouselSchema) => {
-  const { imgSrc, label } = props;
+type CarouselItemImageProps = {
+  imgSrc: string;
+  label: string;
+  index: number;
+  // tinacms accepts any, workaround for better type safety for this component
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  carouselSchema: any;
```
Comment (amankumarrr) [inline components/blocks/carousel.tsx]: maybe we should define some kind of type there to reduce the extra variables coming through.

File: components/events/eventsHeader.tsx [hunk 2]
```diff
@@ -1,14 +1,23 @@
-import Image from "next/image";
+import Image, { getImageProps } from "next/image";
+import { getBackgroundImage } from "../../helpers/images";
```
Comment (amankumarrr) [inline components/events/eventsHeader.tsx:2]: maybe change this to make more meaningful name.

File: components/filter/events.tsx [hunk 3]
```diff
@@ -17,21 +18,51 @@ import { FilterBlock } from "./FilterBlock";
...
+  Url: {
+    Url: string;
+    Description: string;
+  };
+  Presenter?: string;
+  PresenterProfileUrl?: {
+    Url: string;
+  };
+  CalendarType?: string;
+  Category_f5a9cf4c_x002d_8228_x00?: string;
```
Comment (amankumarrr) [inline components/filter/events.tsx:45]: Add a TODO to fix it in the Sharepoint

File: components/filter/events.tsx [hunk 4]
```diff
@@ -74,23 +122,61 @@ const EventTab = ({ children }: { children: React.ReactNode }) => {
...
-              <Event
-                key={index}
-                visible={!!filteredEvents?.find((e) => e.id === event.id)}
-                event={event}
-              />
-            ))
+            events?.map((event, index) => {
+              let eventJsonLd: WithContext<Event> = undefined;
+
+              if (index < 5 && isUpcoming) {
```
Comment (amankumarrr) [inline components/filter/events.tsx]: maybe make this as a constant

File: components/layout/layout.tsx [hunk 5]
```diff
@@ -66,6 +68,24 @@ export const Layout = ({
...
+      case "CLS":
+      case "INP":
+        appInsights.trackMetric(
+          { name: metric.name, average: metric.value },
+          {}
+        );
+        break;
+    }
+  });
+
```
Comment (amankumarrr) [inline components/layout/layout.tsx]: maybe we need to separate it for our main pages (i.e Base pages and Consulting)

File: pages/_app.tsx [hunk 6]
```diff
@@ -24,13 +25,21 @@ import timezone from "dayjs/plugin/timezone";
 import utc from "dayjs/plugin/utc";
 import dynamic from "next/dynamic";
 
+const FIVE_MINS_MS = 1000 * 60 * 5;
```
Comment (amankumarrr) [inline pages/_app.tsx]: maybe make this FIVE_MINS

File: pages/_app.tsx [hunk 7]
```diff
@@ -61,14 +70,29 @@ const App = ({ Component, pageProps }) => {
...
       <AppInsightsProvider>
-        <ErrorBoundary key={router.asPath}>
-          <Component {...pageProps} />
-        </ErrorBoundary>
+        <QueryClientProvider client={queryClient}>
+          <ErrorBoundary key={router.asPath}>
+            <Component {...pageProps} />
+          </ErrorBoundary>
+          {process.env.NODE_ENV === "development" && (
+            <ReactQueryDevtoolsProduction
```
Comment (amankumarrr) [inline pages/_app.tsx]: ReactQueryDevtoolsProduction is not an actual production, maybe change this to something else to reflect its environment

File: components/events/eventsHeader.tsx (hunk 2 above)
Comment (Harry-Ross) [reply components/events/eventsHeader.tsx:2]: i think this is probably ok - see https://nextjs.org/docs/pages/api-reference/components/image#background-css where I've taken the function from

**Review by amankumarrr — APPROVED** [review]
LGTM

**Pull request merged**